tests/spec/test-auth-manager.js

```diff
@@ -0,0 +1,182 @@
+(function () {
+    'use strict';
+
+    var AuthManager = F.manager.AuthManager;
+
+    describe('Auth Manager', function () {
+        var server, token;
+        before(function () {
+            var userInfo = {
+               'jti':'23b6c85b-abcc-443f-93aa-a2bd5e5d4e4b',
+               'sub':'550a2b8b-80f7-4a72-80be-033f87c79cf0',
+               'scope':[
+                  'oauth.approvals',
+                  'openid'
+               ],
+               'client_id':'login',
+               'cid':'login',
+               'grant_type':'password',
+               'user_id':'550a2b8b-80f7-4a72-80be-033f87c79cf0',
+               'user_name':'test_user_in_diff_groups/rippel/',
+               'email':'none@none.com',
+               'iat':1417567152,
+               'exp':1417610352,
+               'iss':'http://localhost:9763/uaa/oauth/token',
+               'aud':[
+                  'oauth',
+                  'openid'
+               ]
+            };
+
+            token = 'eyJhbGciOiJSUzI1NiJ9.' + btoa(JSON.stringify(userInfo)) + '.yYIKw_eWYXAoqPR9aKXs4_';
+            server = sinon.fakeServer.create();
+            server.respondWith(/(.*)\/authentication/, function (xhr, id) {
+                xhr.respond(201, { 'Content-Type': 'application/json' }, JSON.stringify(
+                    { 'refresh_token':'snip-refresh','access_token': token,'expires':43199 }
+                    ));
+            });
+            server.respondWith(/(.*)\/member\/local/, function (xhr, id) {
+                xhr.respond(201, { 'Content-Type': 'application/json' }, JSON.stringify(
+                    { 'refresh_token':'snip-refresh','access_token': token,'expires':43199 }
+                    ));
+            });
+            server.autoRespond = true;
+        });
+
+        after(function () {
+            server.restore();
+            //token = null;
+        });
+
+        it('should pass in transport options to the underlying auth adapter', function () {
+            var callback = sinon.spy();
+            var am = new AuthManager({ transport: { beforeSend: callback } });
+            am.login({ userName: 'john', password: 'y' });
+
+            server.respond();
+            callback.should.have.been.called;
+        });
+
+        describe('#login', function () {
+            it('should login a user without a project', function () {
+                var callback = sinon.spy();
+                var am = new AuthManager();
+                am.login({ userName: 'john', password: 'y', project: null, success: callback });
+                server.respond();
+                // There is a call to the member API inside the sucesss of the first request which makes
+                // the server to not respond immediately, need to investigate if it's possible for the server to respond
+                // in this scenario
+                //callback.should.have.been.called;
+            });
+        });
+
+/*
```

Comment:
This is because we need the mock server to respond within the success callback (the member's api after the auth is successful) and I was not able to fix that in time and was OK'd by Naren.